MathLib/LinAlg/Eigen/EigenMatrix.h

```diff
@@ -76,9 +77,9 @@ class EigenMatrix final
 
     /// add a value to the given entry. If the entry doesn't exist, the value is
     /// inserted.
-    int add(std::size_t row, std::size_t col, double val)
+    int add(IndexType row, IndexType col, double val)
     {
-        assert(row < getNRows() && col < getNCols());
+        assert(row < _mat.rows() && col < _mat.cols());
```

Comment:
I checked EigenMatrix.h code and this assertions are already present there. So I removed our assertions altogether.
See [Eigen/SparseMatrix.h:184](https://bitbucket.org/eigen/eigen/src/25c4b01d1bbce878477809eb86da86d2d2906d53/Eigen/src/SparseCore/SparseMatrix.h?at=default&fileviewer=file-view-default#SparseMatrix.h-184) for example.